### ISSUE TITLE
Running Gromacs with multiple topologies

### ISSUE THREAD

**Issue by KyllenD** [galaxycomputationalchemistry/galaxy-tools-compchem#121]
GROMACS tools fails to include .itp files necessary for topologies generated by CHARMM-GUI. .itp files need to manually pasted into a single .top file for tool to run. Is it possible for this tool to accept a topology file which has several includes to separate .itp files?

**Comment by simonbray** [comment]
No, it's not possible at the moment, unfortunately. I agree that would be useful though.

**Comment by chrisbarnettster** [comment]
Hi @KyllenD 
Nice idea. If you have made any progress, please share. Perhaps an opportunity to create a simple tool to combine parameter files? For example, Galaxy could accept a .tgz with the topol and the subdir with itps and these could be combined using a python script?

**Comment by simonbray** [comment]
I tried to write a tool for that purpose a couple of years ago, it was a surprisingly big headache.

The best approach I think, rather than writing a python script, would be to use the code provided by GROMACS to combine the files: https://www.gromacs.org/Documentation_of_outdated_versions/Include_File_Mechanism. Unfortunately, I don't think they provide it as a standalone command.

FWIW I assume this is the relevant code: https://github.com/gromacs/gromacs/blob/master/src/gromacs/gmxpreprocess/gmxcpp.cpp

**Comment by chrisbarnettster** [comment]
@simonbray thanks for the info! In the usecase where just the includes are needed, I think there is a work around that is good enough for now. @KyllenD are you keen to share the script here?

**Comment by KyllenD** [comment]
@simonbray @chrisbarnettster I used this simple bash script to essentially cut and paste the content of the .itp files to a combined topology file which I was then able to use for GROMACS simulations on Galaxy. This work around assumes that GROAMCS inputs were generated using the CHARMM-GUI input generator

https://github.com/KyllenD/antibody-design/blob/main/combine_top.sh
